app/pulp/app/viewsets/user.py

```diff
@@ -0,0 +1,23 @@
+from pulp.app.models import PulpUser
+from pulp.app.serializers import UserSerializer
+from pulp.app.viewsets import NamedModelViewSet
+
+from rest_framework import decorators, status
+from rest_framework.response import Response
+
+class UserViewSet(NamedModelViewSet):
+    endpoint_name = 'users'
+    queryset = PulpUser.objects.all()
+    serializer_class = UserSerializer
+
+    @decorators.detail_route(methods=['post'])
+    def set_password(self, request, pk=None):
+        user = self.get_object()
+        serializer = UserSerializer(data=request.data)
+        if serializer.is_valid():
```

Comment:
I am *pretty sure* that this will run this validation https://github.com/pulp/pulp/blob/3.0-dev/app/pulp/app/models/auth.py#L48, but I would need to double check.

Another way we could do this is to create a `validate_password` method on the serializer. If named that way, DRF will know about it and do the right thing.